fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -325,53 +327,24 @@ instance IPFS.MonadLocalIPFS Server where
 
 instance IPFS.MonadRemoteIPFS Server where
   runRemote query = do
-    clusterURLs       <- asks ipfsURLs
-    IPFS.Timeout secs <- asks ipfsTimeout
-    manager           <- asks httpManager
+    manager      <- asks ipfsHttpManager
+    urls         <- asks ipfsURLs
+    randomIndex  <- liftIO $ randomRIO (0, NonEmpty.length urls - 1)
+
+    let
+      IPFS.URL url  = urls NonEmpty.Partial.!! randomIndex -- Partial, but checked above
+      clientManager = mkClientEnv manager url
 
-    logDebug @Text "Running IPFS request across cluster"
-    requests <- forM clusterURLs \(IPFS.URL url) ->
-      Process.asyncFor (Seconds (Unity secs)) do
-        runClientM query $ mkClientEnv manager url
+    liftIO $ runClientM query clientManager
 
-    liftIO $ untilDone requests
+instance MonadIPFSCluster Server where
```

Comment:
Do we want to choose a different name for this other than "Cluster" since we're not actually using ipfs-cluster? I can see two possible future confusions:
- we end up using `ipfs-cluster` for something down the line & we now want to delineate between actual interactions with `ipfs-cluster` and interactions with our "group of ipfs nodes"
- someone new hops in the repo and is confused by why the `IPFSCluster` monad doesn't interact with an actual cluster

Probably not a big deal, and both of those could be resolved fairly easily, but just thought I'd mention :man_shrugging:

Reply:
Happy to change the name. It is technically a cluster, but yeah, it's not talking to `ipfs-cluster`. Any name suggestions?

Reply:
Heh yeah I really do like cluster the best, was hoping you'd have an idea :stuck_out_tongue: 

Nothing I can think of is descriptive enough. "swarm" "network" etc all refer to something else, and "group" or "suite", etc aren't as descriptive

So yeah maybe stick with cluster. Idt think the possible future confusion is very severe

Reply:
![Screenshot from 2021-02-22 12-05-13](https://user-images.githubusercontent.com/4898225/108756966-d8aadb80-750e-11eb-94a2-f034a1b3e64f.png)
We could also call it `TheWholeIPFSKitAndCaboodleMonad` :thinking: